### PR TITLE
feat: Enable adding customized logging fields in sensor

### PR DESCRIPTION
**Pull request by emmayylu**
Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

**Comment by emmayylu** [comment]
Partially address https://github.com/argoproj/argo-events/issues/2323 will create other PRs for eventsource and eventbus after this one is merged

**Comment by emmayylu** [comment]
@whynowy Hello Derek! I need this feature for a ticket I am working on at work. This has become my blocker. Could you please review it when you have time? I am very new to Golang, so it would be nice if you could provide some feedback, and I will work on making the PR better! Thank you!

File: pkg/apis/sensor/v1alpha1/types.go [hunk 1]
```diff
@@ -110,6 +110,9 @@ type SensorSpec struct {
 	// RevisionHistoryLimit specifies how many old deployment revisions to retain
 	// +optional
 	RevisionHistoryLimit *int32 `json:"revisionHistoryLimit,omitempty" protobuf:"varint,7,opt,name=revisionHistoryLimit"`
+	// LoggingFields add additional key-value pairs when logging happens
+	// +optional
+	LoggingFields []LoggingField `json:"loggingFields" protobuf:"bytes,8,opt,name=loggingFields"`
```
Comment (whynowy) [inline pkg/apis/sensor/v1alpha1/types.go:115]: Is a `map[string]string` good enough?

use `rep` instead of `opt` in the `json` part

Comment (yolu-kxs) [reply pkg/apis/sensor/v1alpha1/types.go:115]: Thanks for the feedback. I am new to go. I believe if I use `map[string]string`, I won't be able to define the fields (name and value)?

ok! I will change it! I am curious what is the difference between `opt` and `rep`?

Comment (emmayylu) [reply pkg/apis/sensor/v1alpha1/types.go:115]: Thanks for the feedback. I am new to go. I believe if I use `map[string]string`, I won't be able to define the fields (name and value)?

ok! I will change it! I am curious what is the difference between `opt` and `rep`?

Comment (whynowy) [reply pkg/apis/sensor/v1alpha1/types.go]: `rep` means repeated fields in protobuf.


With `map[string]string`, you do
```yaml
  loggingFields:
    a: b
    e: f
```

Comment (yolu-kxs) [reply pkg/apis/sensor/v1alpha1/types.go]: > `rep` means repeated fields in protobuf.
> 
> With `map[string]string`, you do
> 
> ```yaml
>   loggingFields:
>     a: b
>     e: f
> ```

oh I get what you mean now. I was initially thinking to make it look like
```
loggingFields:
- name: job_id
   value: 123
```
and you are suggesting to make it
```
loggingFields:
    job_id: abc
```
Your approach is better

Comment (emmayylu) [reply pkg/apis/sensor/v1alpha1/types.go]: oh I get what you mean now. I was initially thinking to make it look like
```
loggingFields:
- name: job_id
  value: 123
```
and you are suggesting to make it
```
loggingFields:
    job_id: abc
```
Your approach is better

**Review by whynowy — APPROVED** [review]
LGTM

**Pull request merged**